src/components/Editor/Tabs.js

```diff
@@ -21,24 +21,21 @@ const { formatKeyShortcut } = require("../../utils/text");
 require("./Tabs.css");
 
 /*
- * Finds the hidden tabs by comparing the tabs' top offset.
- * hidden tabs will have a great top offset.
+ * Finds the hidden tabs by comparing the tabs' left offset with the right offset of parent.
+ * hidden tabs will have a greater left offset.
  *
  * @param sourceTabs Immutable.list
  * @param sourceTabEls HTMLCollection
  *
  * @returns Immutable.list
  */
-function getHiddenTabs(sourceTabs, sourceTabEls) {
+function getHiddenTabs(sourceTabsParent, sourceTabs, sourceTabEls) {
+  const parentRightOffset = sourceTabsParent.getBoundingClientRect().right;
+  const sourceTabMinWidth = 80; // this value is set in tabs.css
```

Comment:
I think we could have an issue here. With that, we must maintain an implicit sync between this value and the one in the css file.
What would be really nice would be to declare the tab width as a custom property in the CSS file, and then read the value of this custom property here, in JS (see https://www.broken-links.com/2014/08/28/css-variables-updating-custom-properties-javascript/ ).